tests/Tests.AzureAppConfiguration/PushRefreshTests.cs

```diff
@@ -0,0 +1,313 @@
+				EventType = "eventType.KeyValueModified",
+				Id = "some id",
+				Subject = "https://store1.resource.io/kv/searchQuery1",
+				Topic = "/subscriptions/subscription-value1/resourceGroups/resourceGroup1/providers/provider1/configurationstores/store1\\"
+			};
+
+			eventGridEvent.TryCreatePushNotification(out PushNotification pushNotification);
+
+			Assert.Equal(eventGridEvent.EventType, pushNotification.EventType);
+			Assert.Equal(new Uri(eventGridEvent.Subject), pushNotification.ResourceUri);
+			Assert.Equal("syncToken1;sn=001", pushNotification.SyncToken);
+
+			//Should Fail assertions since bad EventGridEvent
+			badEventGridEventDataCapitalization.TryCreatePushNotification(out pushNotification);
+			Assert.False(IsPushNotificationValid(pushNotification));
+		}
+
+        [Fact]
+		public void ProcessPushNotificationThrowsArgumentExceptions()
+        {
+			var mockResponse = new Mock<Response>();
+			var mockClient = GetMockConfigurationClient();
+
+			IConfigurationRefresher refresher = null;
+
+			var config = new ConfigurationBuilder()
+				.AddAzureAppConfiguration(options =>
+				{
+					options.Client = mockClient.Object;
+					options.Select("*");
+					options.ConfigureRefresh(refreshOptions =>
+					{
+						refreshOptions.Register("TestKey1", "label")
+							.SetCacheExpiration(TimeSpan.FromDays(30));
+					});
+					refresher = options.GetRefresher();
+				})
+				.Build();
+
+			foreach (PushNotification invalidPushNotification in invalidPushNotificationList)
+            {
+				Action action = () => refresher.ProcessPushNotification(invalidPushNotification);
+				Assert.Throws<ArgumentException>(action); 
+			}
+
+			PushNotification nullPushNotification = null;
+
+			Action nullAction = () => refresher.ProcessPushNotification(nullPushNotification);
+			Assert.Throws<ArgumentNullException>(nullAction);
+		}
+
+		[Fact]
+		public void SyncTokenUpdatesCorrectNumberOfTimes()
+		{
+			// Arrange
+			var mockResponse = new Mock<Response>();
+			var mockClient = GetMockConfigurationClient();
+
+			IConfigurationRefresher refresher = null;
+
+			var config = new ConfigurationBuilder()
+				.AddAzureAppConfiguration(options =>
+				{
+					options.Client = mockClient.Object;
+					options.Select("*");
+					options.ConfigureRefresh(refreshOptions =>
+					{
+						refreshOptions.Register("TestKey1", "label")
+							.SetCacheExpiration(TimeSpan.FromDays(30));
+					});
+					refresher = options.GetRefresher();
+				})
+				.Build();
+
+			foreach (PushNotification pushNotification in pushNotificationList)
+			{
+				refresher.ProcessPushNotification(pushNotification, TimeSpan.FromSeconds(0));
+				refresher.RefreshAsync().Wait();
+			}
+
+			mockClient.Verify(c => c.GetConfigurationSettingAsync(It.IsAny<ConfigurationSetting>(), It.IsAny<bool>(), It.IsAny<CancellationToken>()), Times.Exactly(8));
+			mockClient.Verify(c => c.UpdateSyncToken(It.IsAny<string>()), Times.Exactly(8));
```

Comment:
Instead of hardcoding 8, we can check if its equal to the length of pushNotificationList. In future, if we add or remove something from the list, this test should still pass.